libpromises/Makefile.am

```diff
@@ -165,29 +165,58 @@ libpromises_la_SOURCES += \
 	process_unix.c \
 	pipes_unix.c
 
+### Process selection.
+#
+# This is significantly OS dependent, so we have per-OS files.
+# A component of the filename reflects this: autotools "PROCESS_OS_TAG".
+#
+# A further complication is a gradual migration from:
+#   : A monolithic "processes_select_cmd" model which use the "ps" command
+# to:
+#   : A JSON-based API with a generic "processes_select_sysprobe"
+#     that calls onto a per-OS "process_sysprobe_/OS/".
+#
+# As the per-OS migration from /cmd/ to /proc/ advances, the /cmd/ files
+# become superfluous and can theoretically be removed.
+#
+# See also "tests/unit/Makefile.am"
+#
+
+if HAVE_PS_VIA_SYSPROBE
+
+### Not yet implemented
+# libpromises_la_SOURCES += \
+#        process_sysprobe_@PROCESS_OS_TAG@.c \
+#        processes_load_sysprobe.c \
+#        processes_select_sysprobe.c
+
+else # HAVE_PS_VIA_SYSPROBE
+
+libpromises_la_SOURCES += processes_select_cmd.c
```

Comment:
All of this is inside a `if !NT` above, so this source will not be added on Windows builds.